<table align="center" width=100%>
    <tr>
        <td width="15%">
            <img src="homework.png">
        </td>
        <td>
            <div align="center">
                <font color="#21618C" size=8px>
                    <b> Take-Home <br>(Day 1)
                    </b>
                </font>
            </div>
        </td>
    </tr>
</table>

### Import the required libraries

In [2]:
import numpy as np
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt
import scipy.stats as stats

#### Read the data

Load the csv file and set the first column as index

In [3]:
df=pd.read_csv("C:/Users/sai/Desktop/New folder (10)/Day 1/car_price.csv",index_col=0)
df.head()

Year  Selling_Price  Present_Price  Kms_Driven Fuel_Type  \
Car_Name                                                             
ritz      2014           3.35           5.59       27000    Petrol   
sx4       2013           4.75           9.54       43000    Diesel   
ciaz      2017           7.25           9.85        6900    Petrol   
wagon r   2011           2.85           4.15        5200    Petrol   
swift     2014           4.60           6.87       42450    Diesel   

         Seller_Type Transmission  Owner  
Car_Name                                  
ritz          Dealer       Manual      0  
sx4           Dealer       Manual      0  
ciaz          Dealer       Manual      0  
wagon r       Dealer       Manual      0  
swift         Dealer       Manual      0

In [5]:
df.shape

(301, 8)

In [6]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Index: 301 entries, ritz to brio
Data columns (total 8 columns):
Year             301 non-null int64
Selling_Price    301 non-null float64
Present_Price    301 non-null float64
Kms_Driven       301 non-null int64
Fuel_Type        301 non-null object
Seller_Type      301 non-null object
Transmission     301 non-null object
Owner            301 non-null int64
dtypes: float64(2), int64(3), object(3)
memory usage: 21.2+ KB


In [5]:
df.isna().sum()

Year             0
Selling_Price    0
Present_Price    0
Kms_Driven       0
Fuel_Type        0
Seller_Type      0
Transmission     0
Owner            0
dtype: int64

In [10]:
df.select_dtypes('object')  # These are the categorical columns in the given dataset.

Fuel_Type Seller_Type Transmission
Car_Name                                   
ritz        Petrol      Dealer       Manual
sx4         Diesel      Dealer       Manual
ciaz        Petrol      Dealer       Manual
wagon r     Petrol      Dealer       Manual
swift       Diesel      Dealer       Manual
...            ...         ...          ...
city        Diesel      Dealer       Manual
brio        Petrol      Dealer       Manual
city        Petrol      Dealer       Manual
city        Diesel      Dealer       Manual
brio        Petrol      Dealer       Manual

[301 rows x 3 columns]

Our objective is to predict the selling price of the cars data.

**The data definition is as follows:** <br><br>
**Car_Name:** name of the car <br>

**YearThis:** year in which the car was bought <br>

**Present_Price:** current ex-showroom price of the car (in lakhs)<br>

**Kms_Driven:** distance completed by the car in km <br>

**Fuel_Type:** fuel type of the car <br>

**Seller_Type:** defines whether the seller is a dealer or an individual<br>

**Transmission:** defines whether the car is manual or automatic <br>

**Owner:** defines the number of owners the car has previously had <br>

**Selling_Price:** price the owner wants to sell the car at (in lakhs) (response variable)

### Let's begin with some hands-on practice exercises

<table align="left">
    <tr>
        <td width="6%">
            <img src="question_icon.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>1. Build a full model and interpret the beta coefficients </b>
                </font>
            </div>
        </td>
    </tr>
</table>

        Hint: A full model is a model which includes all the features 

In [11]:
df = pd.get_dummies(df,drop_first=True)
df.head()

Year  Selling_Price  Present_Price  Kms_Driven  Owner  \
Car_Name                                                          
ritz      2014           3.35           5.59       27000      0   
sx4       2013           4.75           9.54       43000      0   
ciaz      2017           7.25           9.85        6900      0   
wagon r   2011           2.85           4.15        5200      0   
swift     2014           4.60           6.87       42450      0   

          Fuel_Type_Diesel  Fuel_Type_Petrol  Seller_Type_Individual  \
Car_Name                                                               
ritz                     0                 1                       0   
sx4                      1                 0                       0   
ciaz                     0                 1                       0   
wagon r                  0                 1                       0   
swift                    1                 0                       0   

          Transmission_Manual  
Car_Name                       
ritz                        1  
sx4                         1  
ciaz                        1  
wagon r                     1  
swift                       1

In [13]:
# Now biuld the model.

y=df['Selling_Price']
x=df.drop(['Selling_Price'],axis=1)

import statsmodels.api as sm
Xc = sm.add_constant(x)
model = sm.OLS(y,Xc).fit()
model.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:          Selling_Price   R-squared:                       0.883
Model:                            OLS   Adj. R-squared:                  0.879
Method:                 Least Squares   F-statistic:                     274.3
Date:                Tue, 12 Jan 2021   Prob (F-statistic):          5.71e-131
Time:                        16:32:26   Log-Likelihood:                -593.62
No. Observations:                 301   AIC:                             1205.
Df Residuals:                     292   BIC:                             1239.
Df Model:                           8                                         
Covariance Type:            nonrobust                                         
==========================================================================================
                             coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------------------
const                   -789.2366     86.522     -9.122      0.000    -959.522    -618.951
Year                       0.3931      0.043      9.159      0.000       0.309       0.478
Present_Price              0.4372      0.016     27.355      0.000       0.406       0.469
Kms_Driven             -7.013e-06   3.23e-06     -2.172      0.031   -1.34e-05   -6.57e-07
Owner                     -0.6742      0.423     -1.595      0.112      -1.506       0.158
Fuel_Type_Diesel           2.4699      1.278      1.932      0.054      -0.046       4.986
Fuel_Type_Petrol           0.6077      1.259      0.483      0.630      -1.870       3.085
Seller_Type_Individual    -1.1213      0.257     -4.371      0.000      -1.626      -0.616
Transmission_Manual       -1.4482      0.328     -4.417      0.000      -2.093      -0.803
==============================================================================
Omnibus:                       95.107   Durbin-Watson:                   1.795
Prob(Omnibus):                  0.000   Jarque-Bera (JB):              696.392
Skew:                           1.075   Prob(JB):                    6.03e-152
Kurtosis:                      10.135   Cond. No.                     4.56e+07
==============================================================================

Warnings:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
[2] The condition number is large, 4.56e+07. This might indicate that there are
strong multicollinearity or other numerical problems.
"""

Kms_Driven, Seller_Type_Individual, Transmission_Manual are the varaibles with negative coefficent value, which means for 
every 1 unit increase in these columns, there would be some units decrease in the Selling_Price column(Target Variable).

Other variaboes coefficients are positive, which means for every 1 unit increase in these colums would result in some unit increase in the Selling_Price column

<table align="left">
    <tr>
        <td width="6%">
            <img src="question_icon.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>2. Is there multicollinearity present? If yes, which variables are involved in multicollinearity?    </b>
                </font>
            </div>
        </td>
    </tr>
</table>

In [18]:
from statsmodels.stats.outliers_influence import variance_inflation_factor as vif

vif1 = [vif(Xc.values,i) for i in range(Xc.shape[1])]
pd.DataFrame(vif1,index=Xc.columns,columns=['vif_values'])

vif_values
const                   722952.649528
Year                         1.482347
Present_Price                1.836927
Kms_Driven                   1.517916
Owner                        1.056734
Fuel_Type_Diesel            25.182719
Fuel_Type_Petrol            25.026007
Seller_Type_Individual       1.449644
Transmission_Manual          1.196263

Fuel_Type_Diesel and Fuel_Type_Petrol variables havee vif = 25.182719 and 25.026007 which are >10. So, there is a multi-collinearity with these 2-variables.

<table align="left">
    <tr>
        <td width="6%">
            <img src="question_icon.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>3. What is the impact of present price of the car and seller type on the selling price?
                    </b>
                </font>
            </div>
        </td>
    </tr>
</table>

In [24]:
y=df['Selling_Price']
x=df[['Present_Price','Seller_Type_Individual']]

Xc=sm.add_constant(x)
mod = sm.OLS(y,Xc).fit()
mod.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:          Selling_Price   R-squared:                       0.786
Model:                            OLS   Adj. R-squared:                  0.785
Method:                 Least Squares   F-statistic:                     548.4
Date:                Tue, 12 Jan 2021   Prob (F-statistic):          1.34e-100
Time:                        16:49:33   Log-Likelihood:                -683.71
No. Observations:                 301   AIC:                             1373.
Df Residuals:                     298   BIC:                             1385.
Df Model:                           2                                         
Covariance Type:            nonrobust                                         
==========================================================================================
                             coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------------------
const                      1.5423      0.261      5.901      0.000       1.028       2.057
Present_Price              0.4758      0.018     25.956      0.000       0.440       0.512
Seller_Type_Individual    -1.4493      0.331     -4.376      0.000      -2.101      -0.798
==============================================================================
Omnibus:                       72.895   Durbin-Watson:                   1.645
Prob(Omnibus):                  0.000   Jarque-Bera (JB):              845.311
Skew:                           0.581   Prob(JB):                    2.77e-184
Kurtosis:                      11.127   Cond. No.                         33.4
==============================================================================

Warnings:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
"""

<table align="left">
    <tr>
        <td width="6%">
            <img src="question_icon.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>4. Consider all the numeric features in the data. Do all of them significantly contribute to explaining the variation in the selling price?
                    </b>
                </font>
            </div>
        </td>
    </tr>
</table>

In [30]:

y=df['Selling_Price']
x=df[['Present_Price','Kms_Driven','Year']]

Xc=sm.add_constant(x)
mod = sm.OLS(y,Xc).fit()
mod.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:          Selling_Price   R-squared:                       0.850
Model:                            OLS   Adj. R-squared:                  0.849
Method:                 Least Squares   F-statistic:                     561.5
Date:                Tue, 12 Jan 2021   Prob (F-statistic):          5.08e-122
Time:                        16:55:08   Log-Likelihood:                -630.36
No. Observations:                 301   AIC:                             1269.
Df Residuals:                     297   BIC:                             1284.
Df Model:                           3                                         
Covariance Type:            nonrobust                                         
=================================================================================
                    coef    std err          t      P>|t|      [0.025      0.975]
---------------------------------------------------------------------------------
const          -968.4713     93.681    -10.338      0.000   -1152.834    -784.108
Present_Price     0.5256      0.014     38.859      0.000       0.499       0.552
Kms_Driven    -1.214e-06   3.53e-06     -0.344      0.731   -8.15e-06    5.73e-06
Year              0.4813      0.046     10.352      0.000       0.390       0.573
==============================================================================
Omnibus:                       67.241   Durbin-Watson:                   1.524
Prob(Omnibus):                  0.000   Jarque-Bera (JB):             1311.990
Skew:                           0.167   Prob(JB):                    1.27e-285
Kurtosis:                      13.222   Cond. No.                     4.41e+07
==============================================================================

Warnings:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
[2] The condition number is large, 4.41e+07. This might indicate that there are
strong multicollinearity or other numerical problems.
"""

Out of all numerical features, Only Year and Present_Price variables are significant as thier pvalues <0.05. Kms_driven variable is insignificant as its pvalue>0.05.

<table align="left">
    <tr>
        <td width="6%">
            <img src="question_icon.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>5. In the model obtained in question 4, consider the interaction effect of the present price of the car and the year in which it was purchased. Compare the resultant model with the model obtained in previous question and give your interpretation 
                    </b>
                </font>
            </div>
        </td>
    </tr>
</table>

In [34]:
y=df['Selling_Price']
x=df[['Present_Price','Kms_Driven','Year']]

# Adding the intraction column along with independent variables(x)
x['interaction'] = x['Present_Price']*x['Year']

Xc=sm.add_constant(x)
model = sm.OLS(y,Xc).fit()
model.summary ()

C:\Users\narendra\Anaconda3\lib\site-packages\ipykernel_launcher.py:5: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  """


<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:          Selling_Price   R-squared:                       0.963
Model:                            OLS   Adj. R-squared:                  0.963
Method:                 Least Squares   F-statistic:                     1939.
Date:                Tue, 12 Jan 2021   Prob (F-statistic):          6.73e-211
Time:                        17:01:56   Log-Likelihood:                -418.82
No. Observations:                 301   AIC:                             847.6
Df Residuals:                     296   BIC:                             866.2
Df Model:                           4                                         
Covariance Type:            nonrobust                                         
=================================================================================
                    coef    std err          t      P>|t|      [0.025      0.975]
---------------------------------------------------------------------------------
const           100.4679     58.427      1.720      0.087     -14.517     215.452
Present_Price  -137.4867      4.572    -30.068      0.000    -146.485    -128.488
Kms_Driven    -2.142e-06   1.75e-06     -1.224      0.222   -5.59e-06     1.3e-06
Year             -0.0498      0.029     -1.716      0.087      -0.107       0.007
interaction       0.0686      0.002     30.183      0.000       0.064       0.073
==============================================================================
Omnibus:                       36.925   Durbin-Watson:                   1.770
Prob(Omnibus):                  0.000   Jarque-Bera (JB):              177.291
Skew:                          -0.314   Prob(JB):                     3.18e-39
Kurtosis:                       6.707   Cond. No.                     5.73e+07
==============================================================================

Warnings:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
[2] The condition number is large, 5.73e+07. This might indicate that there are
strong multicollinearity or other numerical problems.
"""

Compared to the previous model without interaction, the model with interaction give a way better R-square of 0.963 which is very close to 1. So the model with interaction performs better.

<table align="left">
    <tr>
        <td width="6%">
            <img src="question_icon.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>6. What is the impact of fuel type of cars on the selling price? 
                    </b>
                </font>
            </div>
        </td>
    </tr>
</table>

In [40]:

y=df['Selling_Price']
x=df.iloc[:,[5,6]]

Xc=sm.add_constant(x)
results = sm.OLS(y,Xc).fit()
results.summary()

C:\Users\narendra\Anaconda3\lib\site-packages\numpy\core\fromnumeric.py:2389: FutureWarning: Method .ptp is deprecated and will be removed in a future version. Use numpy.ptp instead.
  return ptp(axis=axis, out=out, **kwargs)
C:\Users\narendra\Anaconda3\lib\site-packages\numpy\core\fromnumeric.py:2389: FutureWarning: Method .ptp is deprecated and will be removed in a future version. Use numpy.ptp instead.
  return ptp(axis=axis, out=out, **kwargs)


<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:          Selling_Price   R-squared:                       0.305
Model:                            OLS   Adj. R-squared:                  0.300
Method:                 Least Squares   F-statistic:                     65.41
Date:                Tue, 12 Jan 2021   Prob (F-statistic):           2.80e-24
Time:                        17:06:33   Log-Likelihood:                -861.21
No. Observations:                 301   AIC:                             1728.
Df Residuals:                     298   BIC:                             1740.
Df Model:                           2                                         
Covariance Type:            nonrobust                                         
====================================================================================
                       coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------------
const                3.1000      3.006      1.031      0.303      -2.816       9.016
Fuel_Type_Diesel     7.1785      3.056      2.349      0.019       1.165      13.192
Fuel_Type_Petrol     0.1642      3.019      0.054      0.957      -5.776       6.105
==============================================================================
Omnibus:                      149.368   Durbin-Watson:                   1.338
Prob(Omnibus):                  0.000   Jarque-Bera (JB):              826.132
Skew:                           2.022   Prob(JB):                    4.05e-180
Kurtosis:                      10.036   Cond. No.                         27.9
==============================================================================

Warnings:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
"""

for the varibles to be significant and to have a impact on target variable, it should have a pvalue<0.05.
Fuel_Type_diesel varaible is significant with pvalue=0.019 and it is impacting the target variable whereas Fuel_Type_Petrol is not-significant.

<table align="left">
    <tr>
        <td width="6%">
            <img src="question_icon.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>7. Does the model significantly explain variation in the target variable? Justify your answer with analysis of variation 
                    </b>
                </font>
            </div>
        </td>
    </tr>
</table>

            Regress the selling price over the transmission.
            
            Selling_Price ~ Transmission

In [43]:
y=df['Selling_Price']
x=df['Transmission_Manual']

Xc=sm.add_constant(x)
result = sm.OLS(y,Xc).fit()
result.summary()

C:\Users\narendra\Anaconda3\lib\site-packages\numpy\core\fromnumeric.py:2389: FutureWarning: Method .ptp is deprecated and will be removed in a future version. Use numpy.ptp instead.
  return ptp(axis=axis, out=out, **kwargs)


<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:          Selling_Price   R-squared:                       0.135
Model:                            OLS   Adj. R-squared:                  0.132
Method:                 Least Squares   F-statistic:                     46.58
Date:                Tue, 12 Jan 2021   Prob (F-statistic):           4.90e-11
Time:                        17:13:19   Log-Likelihood:                -894.20
No. Observations:                 301   AIC:                             1792.
Df Residuals:                     299   BIC:                             1800.
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
=======================================================================================
                          coef    std err          t      P>|t|      [0.025      0.975]
---------------------------------------------------------------------------------------
const                   9.4200      0.749     12.580      0.000       7.946      10.894
Transmission_Manual    -5.4880      0.804     -6.825      0.000      -7.070      -3.906
==============================================================================
Omnibus:                      143.925   Durbin-Watson:                   1.182
Prob(Omnibus):                  0.000   Jarque-Bera (JB):              946.055
Skew:                           1.855   Prob(JB):                    3.69e-206
Kurtosis:                      10.853   Cond. No.                         5.31
==============================================================================

Warnings:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
"""

Pvalue is leas than 0.05, so Transmission effects the Selling Price variable, but the overall model is not good as R-square for overall model = 0.135, which is very less. So, it is recommended to build a model with these 2-variables alone. Include some othre variables also to improve thee performance of the model. 

<table align="left">
    <tr>
        <td width="6%">
            <img src="question_icon.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>8. Regress the selling price over the present price. Compare the 99% and 95% confidence interval of present price of a car
                    </b>
                </font>
            </div>
        </td>
    </tr>
</table>

In [45]:
y=df['Selling_Price']
x=df['Present_Price']

Xc=sm.add_constant(x)
model = sm.OLS(y,Xc).fit()
model.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:          Selling_Price   R-squared:                       0.773
Model:                            OLS   Adj. R-squared:                  0.772
Method:                 Least Squares   F-statistic:                     1016.
Date:                Tue, 12 Jan 2021   Prob (F-statistic):           3.60e-98
Time:                        17:17:10   Log-Likelihood:                -693.08
No. Observations:                 301   AIC:                             1390.
Df Residuals:                     299   BIC:                             1398.
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
=================================================================================
                    coef    std err          t      P>|t|      [0.025      0.975]
---------------------------------------------------------------------------------
const             0.7185      0.187      3.847      0.000       0.351       1.086
Present_Price     0.5168      0.016     31.874      0.000       0.485       0.549
==============================================================================
Omnibus:                       59.775   Durbin-Watson:                   1.533
Prob(Omnibus):                  0.000   Jarque-Bera (JB):              926.121
Skew:                           0.084   Prob(JB):                    7.86e-202
Kurtosis:                      11.592   Cond. No.                         15.4
==============================================================================

Warnings:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
"""

In [51]:
# for 95% Confidence interval(CI), alpha=5%=0.05

model.conf_int(alpha=0.05).tail(1)

0        1
Present_Price  0.484938  0.54876

In [54]:
# for 99% Confidence interval, alpha=1%=0.01

model.conf_int(alpha=0.01).tail(1)

0         1
Present_Price  0.474812  0.558886

The lower limit is high for the 95% CI and the upper limit is high for the 99% CI.

<table align="left">
    <tr>
        <td width="6%">
            <img src="question_icon.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                        <b>9. Verify the statement: The sum of the residuals in any regression model that contains an intercept β<sub>0</sub> is always zero
                    </b>
                </font>
            </div>
        </td>
    </tr>
</table>

        To verify the result, we will fit a regression model of 'Present_Price' on 'Selling_Price' 

In [59]:
y=df['Selling_Price']
x=df['Present_Price']

Xc=sm.add_constant(x)
model = sm.OLS(y,Xc).fit()

residuals=model.resid
print("Sum of the residuals is:",np.sum(residuals))

Sum of the residuals is: -1.1013412404281553e-13


<table align="left">
    <tr>
        <td width="6%">
            <img src="question_icon.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>10. Consider two models as specified below. Compare the performance of the models
                    </b>
                </font>
            </div>
        </td>
    </tr>
</table>

                First model:
        
        Selling_Price ~ Year + Present_Price + Kms_Driven + Owner + Fuel_Type + Seller_Type + Transmission
        
        
                Second model:
        
        Selling_Price ~ Year + Present_Price + Kms_Driven + Owner 

In [69]:
y=df['Selling_Price']
x=df.drop(['Selling_Price'],axis=1)

Xc=sm.add_constant(x)
model1 = sm.OLS(y,Xc).fit()
model1.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:          Selling_Price   R-squared:                       0.883
Model:                            OLS   Adj. R-squared:                  0.879
Method:                 Least Squares   F-statistic:                     274.3
Date:                Tue, 12 Jan 2021   Prob (F-statistic):          5.71e-131
Time:                        17:54:36   Log-Likelihood:                -593.62
No. Observations:                 301   AIC:                             1205.
Df Residuals:                     292   BIC:                             1239.
Df Model:                           8                                         
Covariance Type:            nonrobust                                         
==========================================================================================
                             coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------------------
const                   -789.2366     86.522     -9.122      0.000    -959.522    -618.951
Year                       0.3931      0.043      9.159      0.000       0.309       0.478
Present_Price              0.4372      0.016     27.355      0.000       0.406       0.469
Kms_Driven             -7.013e-06   3.23e-06     -2.172      0.031   -1.34e-05   -6.57e-07
Owner                     -0.6742      0.423     -1.595      0.112      -1.506       0.158
Fuel_Type_Diesel           2.4699      1.278      1.932      0.054      -0.046       4.986
Fuel_Type_Petrol           0.6077      1.259      0.483      0.630      -1.870       3.085
Seller_Type_Individual    -1.1213      0.257     -4.371      0.000      -1.626      -0.616
Transmission_Manual       -1.4482      0.328     -4.417      0.000      -2.093      -0.803
==============================================================================
Omnibus:                       95.107   Durbin-Watson:                   1.795
Prob(Omnibus):                  0.000   Jarque-Bera (JB):              696.392
Skew:                           1.075   Prob(JB):                    6.03e-152
Kurtosis:                      10.135   Cond. No.                     4.56e+07
==============================================================================

Warnings:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
[2] The condition number is large, 4.56e+07. This might indicate that there are
strong multicollinearity or other numerical problems.
"""

In [71]:
y=df['Selling_Price']
x=df[['Year', 'Present_Price', 'Kms_Driven', 'Owner']]

Xc=sm.add_constant(x)
model2 = sm.OLS(y,Xc).fit()
model2.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:          Selling_Price   R-squared:                       0.852
Model:                            OLS   Adj. R-squared:                  0.850
Method:                 Least Squares   F-statistic:                     426.6
Date:                Tue, 12 Jan 2021   Prob (F-statistic):          1.66e-121
Time:                        17:59:11   Log-Likelihood:                -628.25
No. Observations:                 301   AIC:                             1267.
Df Residuals:                     296   BIC:                             1285.
Df Model:                           4                                         
Covariance Type:            nonrobust                                         
=================================================================================
                    coef    std err          t      P>|t|      [0.025      0.975]
---------------------------------------------------------------------------------
const          -937.7642     94.392     -9.935      0.000   -1123.528    -752.000
Year              0.4661      0.047      9.949      0.000       0.374       0.558
Present_Price     0.5256      0.013     39.067      0.000       0.499       0.552
Kms_Driven    -1.267e-06   3.51e-06     -0.361      0.718   -8.17e-06    5.64e-06
Owner            -0.9513      0.466     -2.042      0.042      -1.868      -0.034
==============================================================================
Omnibus:                       69.450   Durbin-Watson:                   1.560
Prob(Omnibus):                  0.000   Jarque-Bera (JB):             1362.873
Skew:                           0.241   Prob(JB):                    1.14e-296
Kurtosis:                      13.413   Cond. No.                     4.46e+07
==============================================================================

Warnings:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
[2] The condition number is large, 4.46e+07. This might indicate that there are
strong multicollinearity or other numerical problems.
"""

For first model, R-squared= 0.883 and for second model R-squared=0.852. Hence, first model is better as it's r-square is little high  compared to second model.